library/ssl_tls13_generic.c

```diff
@@ -296,51 +375,33 @@ static int ssl_tls13_parse_certificate_verify( mbedtls_ssl_context *ssl,
     p += 2;
     MBEDTLS_SSL_CHK_BUF_READ_PTR( p, end, signature_len );
 
-    /* Hash verify buffer with indicated hash function */
-    switch( md_alg )
+    status = psa_hash_compute( hash_alg,
+                               verify_buffer,
+                               verify_buffer_len,
+                               verify_hash,
+                               PSA_HASH_LENGTH( hash_alg ),
```

Comment:
The output size parameter indicates how much room there is in the buffer. It has no relation to how much data you expect from the function.
```suggestion
                               sizeof( verify_hash ),
```